tests/integration/test_increment_graph_window.py

```diff
@@ -0,0 +1,410 @@
+            # the blink task is n=4
+        },
+    }
+
+    def advance():
+        """Advance to the next task in the workflow.
+
+        This works its way down the chain of tasks between "a" and "s"
+        inclusive, yielding what the n-window should look like for this
+        workflow at each step.
+
+        Yields:
+            tuple - (previous_task, active_task, n_window)
+
+            previous_task:
+                The task which has just "succeeded".
+            active_task:
+                The task which is about to run.
+            n_window:
+                Dictionary of {task_name: graph_depth} for the n=3 window.
+
+        """
+        # the initial window on startup (minus the nodes traversed via "blink")
+        window = {
+            'a': 0,
+            'b': 1,
+            'c': 2,
+            'd': 3,
+        }
+        # the tasks we will run in order
+        letters = 'abcdefghijklmnopqrs'
+        # the graph-depth of the "blink" task at each stage of the workflow
+        blink_distances = [1] + (list((*range(2, 5), *range(3, 0, -1))) * 3)
+
+        for ind, blink_distance in zip(range(len(letters)), blink_distances):
+            previous_task = letters[ind - 1] if ind > 0 else None
+            active_task = letters[ind]
+            yield (
+                previous_task,
+                active_task,
+                {
+                    # the tasks traversed via the "blink" task
+                    **blink[blink_distance],
+                    # the tasks in the main "abcdefg" chain
+                    **{key: abs(value) for key, value in window.items()},
+                }
+            )
+
+            # move each task in the "abcdef" chain down one
+            window = {key: value - 1 for key, value in window.items()}
+            # add the n=3 task in the "abcdef" chain into the window
+            with suppress(IndexError):
+                window[letters[ind + 4]] = 3
+            # pull out anything which is not supposed to be in the n=3 window
+            window = {
+                key: value
+                for key, value in window.items()
+                if abs(value) < 4
+            }
+
+    async with start(schd):
+        schd.data_store_mgr.set_graph_window_extent(3)
+        await schd.update_data_structure()
+
+        previous_n_window = {}
+        for previous_task, active_task, n_window in advance():
+            # mark the previous task as completed
+            await complete_task(schd, previous_task)
+            # add the next task to the pool
+            add_task(schd, active_task)
+            # run the graph window algorithm
+            increment_graph_window(schd, active_task)
+            # get the deltas which increment_graph_window created
+            added, updated, pruned = get_deltas(schd)
+
+            # compare the n-window in the store to what we were expecting
+            _n_window = await get_n_window(schd)
+            assert _n_window == n_window
```

Comment:
I think it would be clearer if `n_window -> expected_n_window` and `_n_window -> n_window` or just
```suggestion
            assert (await get_n_window(schd)) == n_window
```